File: coding-test-1/practice-problems/solutions/05-count-vowels.rb
# Write a method that takes a string and returns the number of vowels
# in the string. You may assume that all the letters are lower cased.
# You can treat "y" as a consonant.
#
# Difficulty: easy.

def count_vowels(string)
  num_vowels = 0

  i = 0
  while i < string.length
    if (string[i] == "a" || string[i] == "e" || string[i] == "i" || string[i] == "o" || string[i] == "u")
      num_vowels += 1
    end

    i += 1
  end

  return num_vowels
end

# These are tests to check that your code is working. After writing
# your solution, they should all print true.

puts('count_vowels("abcd") == 1: ' + (count_vowels('abcd') == 1).to_s)
puts('count_vowels("color") == 2: ' + (count_vowels('color') == 2).to_s)
puts('count_vowels("colour") == 3: ' + (count_vowels('colour') == 3).to_s)
puts('count_vowels("cecilia") == 4: ' + (count_vowels('cecilia') == 4).to_s)

File: coding-test-1/practice-problems/solutions-python/05-count-vowels.py
# Write a method that takes a string and returns the number of vowels
# in the string. You may assume that all the letters are lower cased.
# You can treat "y" as a consonant.
#
# Difficulty: easy.
import unittest


def count_vowels(string):
    vowels = "aeiou"
    vowel_count = 0
    for letter in list(string):
        if letter in vowels:
            vowel_count += 1
    return vowel_count


# Check out list comprehensions
# def count_vowels_v2(string):
#     return sum([1 for letter in list(string) if letter in "aeiou"])


# These are tests to check that your code is working. After writing
# your solution, they should all run.

class CountVowels(unittest.TestCase):

    def test_abcd(self):
        self.assertEqual(count_vowels("abcd"), 1)

    def test_word_color(self):
        self.assertEqual(count_vowels("color"), 2)

    def test_fancy_color(self):
        self.assertEqual(count_vowels("colour"), 3)

    def test_jane_doe(self):
        self.assertEqual(count_vowels("cecilia"), 4)

if __name__ == '__main__':
    unittest.main()